### ISSUE TITLE
iOS 12.2 crash - 'NSInvalidArgumentException', reason: 'Unsupported fetch request

### ISSUE THREAD

**Issue by abdelrahman-badary** [GetStream/stream-chat-swift#1414]
## What did you do?
Hello, I am using the `StreamChatUI` version `4.0.0-beta.11` when running on iOS 12 (12.3.1 and 12.2 specifically cause I haven't tried on other 12 versions) the application crashes the moment I open any screen that contains the chat module.

## Stack trace 
The crash breakpoints stops in the `MessageDTO.swift` file in the method `loadAttachmentCounts` while executing this statement `return try context.fetch(request).reduce(into: [:])`
here is the full stack trace :

`*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Unsupported fetch request: store <NSMemoryObjectStore: 0x7f82281a7da0> (URL: file:///dev/null) does not support GROUP BY'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000110c0d6fb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x00000001164edac5 objc_exception_throw + 48
	2   CoreData                            0x0000000115fe3f7e -[NSMappedObjectStore executeFetchRequest:withContext:] + 3294
	3   CoreData                            0x0000000115fe31fc -[NSMappedObjectStore executeRequest:withContext:error:] + 364
	4   CoreData                            0x0000000116094007 __65-[NSPersistentStoreCoordinator executeRequest:withContext:error:]_block_invoke + 2039
	5   CoreData                            0x000000011608c35e __55-[NSPersistentStoreCoordinator _routeHeavyweightBlock:]_block_invoke + 62
	6   CoreData                            0x00000001160a0ab4 gutsOfBlockToNSPersistentStoreCoordinatorPerform + 212
	7   libdispatch.dylib                   0x000000011949edb5 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x00000001194acd08 _dispatch_lane_barrier_sync_invoke_and_complete + 132
	9   CoreData                            0x000000011608bec5 _perform + 213
	10  CoreData                            0x000000011608c1d6 -[NSPersistentStoreCoordinator _routeHeavyweightBlock:] + 214
	11  CoreData                            0x0000000115f84fc0 -[NSPersistentStoreCoordinator executeRequest:withContext:error:] + 1296
	12  CoreData                            0x0000000115f83295 -[NSManagedObjectContext executeFetchRequest:error:] + 933
	13  libswiftCoreData.dylib              0x0000000118e7d71f $sSo22NSManagedObjectContextC8CoreDataE5fetchySayxGSo14NSFetchRequestCyxGKSo0gH6ResultRzlF + 63
	14  Eventtus                            0x0000000107092e9d $s10StreamChat10MessageDTOC20loadAttachmentCounts3for7contextSDyAA0F4TypeVSiGSS_So22NSManagedObjectContextCtFZTf4nnd_n + 1165
	15  Eventtus                            0x0000000107091904 $sSDy10StreamChat14AttachmentTypeVSiGIego_ADIegr_TRTA + 20
	16  Eventtus                            0x0000000106f65938 $s10StreamChat12CoreDataLazyC12wrappedValuexvgyxSgzXEfU_yycfU_SDyAA14AttachmentTypeVSiG_Tg5 + 232
	17  Eventtus                            0x0000000107016dd3 $s10StreamChat12CoreDataLazyC12wrappedValuexvgyxSgzXEfU_yyXEfU0_ + 19
	18  Eventtus                            0x000000010573782c $sIg_Ieg_TR + 12
	19  Eventtus                            0x0000000105737851 $sIeg_IyB_TR + 17
	20  CoreData                            0x0000000115fc684a developerSubmittedBlockToNSManagedObjectContextPerform + 170
	21  CoreData                            0x0000000115fc671f -[NSManagedObjectContext performBlockAndWait:] + 239
	22  Eventtus                            0x0000000106f64e37 $s10StreamChat12CoreDataLazyC12wrappedValuexvgyxSgzXEfU_SDyAA14AttachmentTypeVSiG_Tg5 + 407
	23  Eventtus                            0x0000000106f62e4b $s10StreamChat12CoreDataLazyC12wrappedValuexvgSayAA0B13ChannelMemberCG_Tg5Tm + 139
	24  Eventtus                            0x0000000106fe39aa $s10StreamChat0B7MessageV16attachmentCountsSDyAA14AttachmentTypeVSiGvg + 42
	25  Eventtus                            0x0000000107139201 $s12StreamChatUI21AttachmentViewCatalogC010attachmentE16InjectorClassFor7message10componentsAA0deH0CmSg0aB00B7MessageV_AA10ComponentsVtFZ + 129
	26  Eventtus                            0x000000010719c872 $s12StreamChatUI0B13MessageListVCC030attachmentViewInjectorClassForD02atAA010AttachmenthI0CmSg10Foundation9IndexPathV_tF + 146
	27  Eventtus                            0x00000001059c1665 $s8Eventtus19BVChatMessageListVCC9tableView_12cellForRowAtSo07UITableG4CellCSo0lG0C_10Foundation9IndexPathVtF + 661
	28  Eventtus                            0x00000001059c19b5 $s8Eventtus19BVChatMessageListVCC9tableView_12cellForRowAtSo07UITableG4CellCSo0lG0C_10Foundation9IndexPathVtFTo + 165
	29  UIKitCore                           0x000000012106662c -[UITableView _createPreparedCellForGlobalRow:withIndexPath:willDisplay:] + 764
	30  UIKitCore                           0x0000000121066b65 -[UITableView _createPreparedCellForGlobalRow:willDisplay:] + 73
	31  UIKitCore                           0x000000012102ed20 -[UITableView _updateVisibleCellsNow:isRecursive:] + 2870
	32  UIKitCore                           0x000000012104ee37 -[UITableView layoutSubviews] + 165
	33  UIKitCore                           0x00000001212fb9c1 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1417
	34  QuartzCore                          0x00000001143fbeae -[CALayer layoutSublayers] + 173
	35  QuartzCore                          0x0000000114400b88 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 396
	36  UIKitCore                           0x00000001212e6985 -[UIView(Hierarchy) layoutBelowIfNeeded] + 646
	37  Eventtus                            0x0000000107199e68 $s12StreamChatUI0B13MessageListVCC24traitCollectionDidChangeyySo07UITraitH0CSgFTo + 136
	38  UIKitCore                           0x0000000120829561 -[UIViewController _traitCollectionDidChange:] + 108
	39  UIKitCore                           0x0000000120829684 -[UIViewController _updateTraitsIfNecessary] + 269
	40  UIKitCore                           0x00000001212fb80f -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 983
	41  QuartzCore                          0x00000001143fbeae -[CALayer layoutSublayers] + 173
	42  QuartzCore                          0x0000000114400b88 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 396
	43  QuartzCore                          0x000000011440cee4 _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 72
	44  QuartzCore                          0x000000011437c3aa _ZN2CA7Context18commit_transactionEPNS_11TransactionE + 328
	45  QuartzCore                          0x00000001143b3584 _ZN2CA11Transaction6commitEv + 608
	46  QuartzCore                          0x00000001143b3ede _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 76
	47  CoreFoundation                      0x0000000110b740f7 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	48  CoreFoundation                      0x0000000110b6e5be __CFRunLoopDoObservers + 430
	49  CoreFoundation                      0x0000000110b6ec31 __CFRunLoopRun + 1505
	50  CoreFoundation                      0x0000000110b6e302 CFRunLoopRunSpecific + 626
	51  GraphicsServices                    0x000000011a1d82fe GSEventRunModal + 65
	52  UIKitCore                           0x0000000120e2dba2 UIApplicationMain + 140
	53  Eventtus                            0x000000010675e81b main + 75
	54  libdyld.dylib                       0x0000000119513541 start + 1
)
2021-09-01 12:19:20.489740+0200 Eventtus[55388:2171609] TIC Read Status [16:0x7f82299a5fb0]: 1:57
2021-09-01 12:19:20.506834+0200 Eventtus[55388:2171609] [BoringSSL] nw_protocol_boringssl_error(1584) [C16.1:2][0x7f8229981b00] Lower protocol stack error: 32
libc++abi.dylib: terminating with uncaught exception of type NSException`

## GetStream Environment

**GetStream Chat version:**
`StreamChatUI` version `4.0.0-beta.11`
**GetStream Chat frameworks:** StreamChatUI
**iOS version:** iOS 12
**Swift version:** Swift 5
**Xcode version:** 12.5
**Device:** Any device with iOS 12 on it

**Comment by tbarbugli** [comment]
can you share the code that you use to setup StreamChat?

**Comment by abdelrahman-badary** [comment]
@tbarbugli which part of the code ? the connection to GetStream client itself or the part displaying the chat components?

**Comment by tbarbugli** [comment]
ideally as much as possible but at least: 

- the code used to init the ChatClient and connect the user 
- the code used to display the VC that is crashing

please make sure to not include any secret in there like user tokens or api secret ;)

**Comment by abdelrahman-badary** [comment]
@tbarbugli Here is what I can share.

**This the method that inits the ChatClient and connects the user**

```
func connectToGetStream(userId: String, userToken: String, gggxKey: String) { 

        let chatClientConfig = ChatClientConfig(apiKey: .init(gggxKey))
        // Here chatClient is a class property.
        self.chatClient = ChatClient(config: chatClientConfig, tokenProvider: { [weak self] compeletion in
            guard let strongSelf = self else { return }
            strongSelf.refreshGetStreamToken(compeletion: compeletion)
        })
        guard let token = try? Token.init(rawValue: userToken) else { return }
        chatClient?.connectUser(userInfo: .init(id: userId), token: token, completion: { error in
        })
    }
```

**This is the method that gets the ChatController from that chat client**
```
    func getChatControllerFor(channelId: String, type: CustomChatChannelType, compeletion: @escaping (ChatChannelController?) -> ()) {
        let channelId = ChannelId(type: .custom(type.rawValue), id: channelId)
        let chatController = chatClient?.channelController(for: channelId)
        chatController?.synchronize({ error in
            compeletion(chatController)
        })
    }
```
The 2 methods above are inside a singleton class that is initialized in the app delegate, the method `connectToGetStream` is also called in the app delegate, the method `getChatControllerFor` gets called for each view that displays the chat components.

**This is the method that display the chat view**

```
    func addChatView() {

        let messageListVc = ChatMessageListVC()
        // getStreamClient is a object from the Singleton class that handles ChatClient connection
        getStreamClient.getChatControllerFor(channelId: chatChannelId, type: self.chatType) { [weak self] channelController in
            guard let strongSelf = self else { return }
            strongSelf.channelController = channelController
            strongSelf.eventsController = channelController?.eventsController()
            strongSelf.eventsController?.delegate = strongSelf
            messageListVc.channelController = channelController
            strongSelf.childWillMove(child: messageListVc)
            messageListVc.willMove(toParent: strongSelf)
            strongSelf.addChild(messageListVc)
            messageListVc.view.translatesAutoresizingMaskIntoConstraints = false
            strongSelf.chatContainer.addSubview(messageListVc.view)
            strongSelf.chatContainer.pinItemToEdges(item: messageListVc.view)
            messageListVc.didMove(toParent: strongSelf)
        }
    }
```
The above method is a part of a `UIViewController` that holds a view that displays the chat.

**Comment by tbarbugli** [comment]
#1415 should fix this problem, we'll get this one merged soon. Keep in mind that that branch is from latest main which has a couple important changes around `ChatMessageListVC` (https://github.com/GetStream/stream-chat-swift/blob/main/CHANGELOG.md#%EF%B8%8F-breaking-changes-from-40-beta11)

**Comment by abdelrahman-badary** [comment]
Thanks @tbarbugli for the fast recovery, do we have a date for the new beta version?

**Comment by tbarbugli** [comment]
tomorrow we'll release it :)

**Comment by tbarbugli** [comment]
this issue should be resolved on the latest release (4.0-rc1) please reopen if you can still reproduce

**Comment by abdelrahman-badary** [comment]
@tbarbugli I tested it I can confirm it is resolved
